src/mem_table.rs

```diff
@@ -110,26 +117,27 @@ impl WritableKVTable {
         &self.table
     }
 
-    pub(crate) fn put(&mut self, key: &[u8], value: &[u8]) {
+    pub(crate) fn put(&mut self, key: &[u8], value: &[u8], attrs: RowAttributes) {
         self.maybe_subtract_old_val_from_size(key);
-        self.size = self.size + key.len() + value.len();
-        self.table.put(key, value)
+        self.size = self.size + key.len() + value.len() + sizeof_attributes(&attrs);
+        self.table.put(key, value, attrs)
     }
 
-    pub(crate) fn delete(&mut self, key: &[u8]) {
+    pub(crate) fn delete(&mut self, key: &[u8], attrs: RowAttributes) {
```

Comment:
can we still delete a key if we dunno the `RowAttributes` on the caller sites?

Reply:
sorry plz ignore my previous comment, i noticed it will delete the entry in the memtable by the key, and in the current design, a memtable will always store the latest value of a key. am i understand it correctly?

Reply:
Yes, that is correct